### DISCUSSION TITLE
Dual Modules Port Names

### DISCUSSION THREAD

**Discussion by hobbit666** [netbox-community/netbox#15534] [Help Wanted!]
Did have a quick search but couldn't find the answer.

I've created a device type for a server and added 2 module bays for dual power supply's
I then created a device type call Dell 750 PSU with a Power Port called PW1
When i create the device and add the modules, when i add the 2nd PSU it gives me a message that power port name is already in use.

I'm after some advice on how to handle when device have redundant module, for expamle i will be adding a Dell SC storage device that will have 2 controllers each with 4 lan ports.  So if i create a module type e.g.  Dell Controller with 4 interfaces eth1 eth2 etc.  When i come to add them to the device i'll get the same issue,

**Comment by a084ed22** [comment]
Considering that each controller is independent and has its own management layer, you'd probably want to create a container device with two 0-unit devices for the controllers. Unfortunately, properly mapping the shared nature of other parts i.e. fans, PSUs, etc, doesn't map nicely in netbox and you'll have to make some choices in how to treat them.

You may also want to look at the [device type library](https://github.com/netbox-community/devicetype-library) for references and examples about how the problem may have already been solved by others

**Comment by candlerb** [comment]
In the power port template's name use the string `{module}` literally, e.g. `PW{module}` - then each component will get a unique name based on the module name.
